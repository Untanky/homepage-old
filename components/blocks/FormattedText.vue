<script lang="ts">
import Vue, { PropType } from 'vue';
import FormattedText from '../../src/components/FormattedText';

export default Vue.extend({
  name: 'FormattedText',
  props: {
    data: {
      type: Object as PropType<FormattedText>,
      required: true,
    },
  },
  methods: {
    elementType() {
      if (!this.data.variant || this.data.variant === '') {
        return 'div';
      }

      return this.data.variant;
    },
    classes() {
      const classList = [...this.data.classes];

      if ([undefined, '', 'div', 'span', 'code'].includes(this.data.variant) && this.data.size) {
        classList.push(`text-${this.data.size}`);
      }

      if (this.data.color) {
        classList.push(`text-${this.data.color}`);
      }

      if (this.data.alignment) {
        classList.push(`text-${this.data.alignment}`);
      }

      if (this.data.bold) {
        classList.push('font-bold');
      }

      if (this.data.italic) {
        classList.push('italic');
      }

      if (this.data.underline) {
        classList.push('underline');
      }

      return classList;
    },
  },
  render(createElement) {
    console.log(Object.keys(this));
    return createElement(
      this.elementType(),
      {
        class: this.classes(),
      },
      [
        // eslint-disable-next-line no-underscore-dangle
        this._v(this.data.text),
      ],
    );
  },
});
</script>
