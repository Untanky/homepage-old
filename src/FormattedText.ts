type FormattedText = {
  text: string,
  classes: string[],
}

export default FormattedText;
