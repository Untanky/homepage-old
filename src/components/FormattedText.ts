import ComponentData from "../ComponentData";

interface FormattedText extends ComponentData {
  text: string,
  classes: string[],
}

export default FormattedText;
