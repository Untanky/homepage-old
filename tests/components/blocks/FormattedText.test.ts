/* eslint-disable no-undef */
// global it, expect
import { mount } from '@vue/test-utils';
import FormattedText from '../../../components/blocks/FormattedText.vue';

describe('FormattedText', () => {
  it('should run without config', () => {
    const props = {
      data: {
        text: 'Hello World',
        classes: [],
      },
    };

    const comp = mount(FormattedText, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run as header', () => {
    const props = {
      data: {
        text: 'Hello World',
        variant: 'h1',
        classes: [],
      },
    };

    const comp = mount(FormattedText, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run as code snippet', () => {
    const props = {
      data: {
        text: 'Hello World',
        variant: 'code',
        classes: [],
      },
    };

    const comp = mount(FormattedText, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run with size', () => {
    const props = {
      data: {
        text: 'Hello World',
        size: 'xl',
        classes: [],
      },
    };

    const comp = mount(FormattedText, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run with size and header variant', () => {
    const props = {
      data: {
        text: 'Hello World',
        variant: 'h1',
        size: 'xl',
        classes: [],
      },
    };

    const comp = mount(FormattedText, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run with size and normal variant', () => {
    const props = {
      data: {
        text: 'Hello World',
        variant: 'div',
        size: 'xl',
        classes: [],
      },
    };

    const comp = mount(FormattedText, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run with color red', () => {
    const props = {
      data: {
        text: 'Hello World',
        color: 'red-500',
        classes: [],
      },
    };

    const comp = mount(FormattedText, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run with alignment left', () => {
    const props = {
      data: {
        text: 'Hello World',
        alignment: 'left',
        classes: [],
      },
    };

    const comp = mount(FormattedText, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run with alignment center', () => {
    const props = {
      data: {
        text: 'Hello World',
        alignment: 'center',
        classes: [],
      },
    };

    const comp = mount(FormattedText, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run with alignment right', () => {
    const props = {
      data: {
        text: 'Hello World',
        alignment: 'right',
        classes: [],
      },
    };

    const comp = mount(FormattedText, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run with alignment justify', () => {
    const props = {
      data: {
        text: 'Hello World',
        alignment: 'justify',
        classes: [],
      },
    };

    const comp = mount(FormattedText, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run with bold font', () => {
    const props = {
      data: {
        text: 'Hello World',
        bold: true,
        classes: [],
      },
    };

    const comp = mount(FormattedText, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run with italic font', () => {
    const props = {
      data: {
        text: 'Hello World',
        italic: true,
        classes: [],
      },
    };

    const comp = mount(FormattedText, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run when underlined', () => {
    const props = {
      data: {
        text: 'Hello World',
        underline: true,
        classes: [],
      },
    };

    const comp = mount(FormattedText, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run with classes', () => {
    const props = {
      data: {
        text: 'Hello World',
        classes: ['custom-class-1', 'test-class'],
      },
    };

    const comp = mount(FormattedText, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run when everything configured', () => {
    const props = {
      data: {
        text: 'Hello World',
        variant: 'h2',
        color: 'green-500',
        alignment: 'center',
        bold: true,
        italic: true,
        underline: true,
        classes: [],
      },
    };

    const comp = mount(FormattedText, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });
});
