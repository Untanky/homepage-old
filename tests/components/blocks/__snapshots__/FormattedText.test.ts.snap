// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormattedText should run as code snippet 1`] = `
<code
  class=""
>
  Hello World
</code>
`;

exports[`FormattedText should run as header 1`] = `
<h1
  class=""
>
  Hello World
</h1>
`;

exports[`FormattedText should run when everything configured 1`] = `
<h2
  class="text-green-500 text-center font-bold italic underline"
>
  Hello World
</h2>
`;

exports[`FormattedText should run when underlined 1`] = `
<div
  class="underline"
>
  Hello World
</div>
`;

exports[`FormattedText should run with alignment center 1`] = `
<div
  class="text-center"
>
  Hello World
</div>
`;

exports[`FormattedText should run with alignment justify 1`] = `
<div
  class="text-justify"
>
  Hello World
</div>
`;

exports[`FormattedText should run with alignment left 1`] = `
<div
  class="text-left"
>
  Hello World
</div>
`;

exports[`FormattedText should run with alignment right 1`] = `
<div
  class="text-right"
>
  Hello World
</div>
`;

exports[`FormattedText should run with bold font 1`] = `
<div
  class="font-bold"
>
  Hello World
</div>
`;

exports[`FormattedText should run with classes 1`] = `
<div
  class="custom-class-1 test-class"
>
  Hello World
</div>
`;

exports[`FormattedText should run with color red 1`] = `
<div
  class="text-red-500"
>
  Hello World
</div>
`;

exports[`FormattedText should run with italic font 1`] = `
<div
  class="italic"
>
  Hello World
</div>
`;

exports[`FormattedText should run with size 1`] = `
<div
  class="text-xl"
>
  Hello World
</div>
`;

exports[`FormattedText should run with size and header variant 1`] = `
<h1
  class=""
>
  Hello World
</h1>
`;

exports[`FormattedText should run with size and normal variant 1`] = `
<div
  class="text-xl"
>
  Hello World
</div>
`;

exports[`FormattedText should run without config 1`] = `
<div
  class=""
>
  Hello World
</div>
`;
